pkgs/shells/fish/plugins/pure.nix

```diff
@@ -1,29 +1,21 @@
-{ lib, buildFishPlugin, fetchFromGitHub, git, fishtape }:
+{ lib, buildFishPlugin, fetchFromGitHub, git, fishtape_3 }:
 
 buildFishPlugin rec {
   pname = "pure";
-  version = "4.7.0";
+  version = "4.8.0";
 
   src = fetchFromGitHub {
     owner = "pure-fish";
     repo = "pure";
     rev = "v${version}";
-    hash = "sha256-2ZNb6aB7rIubTfRVRb42xmCdImQmtGGubo0TBwSPHEo=";
+    hash = "sha256-O8rC2uCuM3xUQPRap7XqyyAvO77hP+sqNM4mEQ7pZkw=";
   };
 
-  # The tests aren't passing either on the project's CI.
-  # The release notes of the program for v3.5.0 say:
-  # > Tests are going crazy at the moment, should be fixed once fishtape 3.0
-  # > is released, and we do the switch.
-  # This is tracked in https://github.com/pure-fish/pure/issues/272
-  # and https://github.com/pure-fish/pure/pull/275.
-  doCheck = false;
-
   nativeCheckInputs = [ git ];
-  checkPlugins = [ fishtape ];
+  checkPlugins = [ fishtape_3 ];
   checkPhase = ''
-    # https://github.com/rafaelrinaldi/pure/issues/264
```

Comment:
Put the `runHook`s

Reply:
There's no such thing for the `checkPhase` of `buildFishPlugin`?
That might perhaps be missing in `build-fish-plugin.nix` itself.

Reply:
They need to be added manually when you override the default `checkPhase` with custom code.

Reply:
Though `buildFishPlugin`'s `checkPhase` is "wrapped" here to be executed with Fish instead of Bash:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/shells/fish/plugins/build-fish-plugin.nix#L73

Using `runHook` here drops to Bash again?
So the `runHook`s might be missing there and not in this package.
(That attribute should perhaps be renamed into `checkScript` too, since it's behaving differently from the rest).

Reply:
For the record, adding the `runHook` here results in the following:

```log
running tests
fish: Unknown command: runHook
/nix/store/i93fi9jmzikv4kf6r6xscdf3v763jl03-fishplugin-pure-4.7.0-test (line 1):
runHook preCheck
^~~~~~^
[test...]
fish: Unknown command: runHook
/nix/store/i93fi9jmzikv4kf6r6xscdf3v763jl03-fishplugin-pure-4.7.0-test (line 8):
runHook postCheck
^~~~~~^
```

Reply:
This should be treated as a bug, then.

Let's ignore for now.